git/pre-commit

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+if [ "$NOLINT" ]; then exit 0; fi
+
+set -v
+
+FLAKE8_CONFIG_DIR="$(dirname $0)/cfg"
+
+flake8 . --config=${FLAKE8_CONFIG_DIR}/travis_run_flake8__init__.cfg
+status1=$?
+flake8 . --config=${FLAKE8_CONFIG_DIR}/travis_run_flake8.cfg
+status2=$?
+
+set -v
+
+PYLINT_CONFIG_DIR="$(dirname $0)/cfg"
+# In the structure, hooks are in .git/hooks, so we need to go 2 levels up
+GIT_DIR="$(dirname $(dirname $(dirname $0)))"
+
+# Fix pylint path. More info: https://www.mail-archive.com/code-quality@python.org/msg00294.html
+export PYTHONPATH=${PYTHONPATH}:$GIT_DIR
+touch $GIT_DIR/__init__.py
+
+# run pylint command - For some strange reason, message controls are not taken correctly from cfg
+pylint --rcfile=${PYLINT_CONFIG_DIR}/travis_run_pylint.cfg --disable=all --enable=E0101,E1124,E1306,E1601,I0013,W0101,W0102,W0104,W0105,W0109,W0403,W0404,W1111,W1401 $GIT_DIR
```

Comment:
yes I did.
pylint 1.4.1
astroid-1.3.4



On Mon, Mar 2, 2015 at 1:16 PM, Pedro M. Baeza <notifications@github.com>
wrote:

> In git/pre-commit
> <https://github.com/OCA/maintainer-quality-tools/pull/166#discussion_r25594148>
> :
>
> > +status1=$?
> > +flake8 . --config=${FLAKE8_CONFIG_DIR}/travis_run_flake8.cfg
> > +status2=$?
> > +
> > +set -v
> > +
> > +PYLINT_CONFIG_DIR="$(dirname $0)/cfg"
> > +# In the structure, hooks are in .git/hooks, so we need to go 2 levels up
> > +GIT_DIR="$(dirname $(dirname $(dirname $0)))"
> > +
> > +# Fix pylint path. More info: https://www.mail-archive.com/code-quality@python.org/msg00294.html
> > +export PYTHONPATH=${PYTHONPATH}:$GIT_DIR
> > +touch $GIT_DIR/__init__.py
> > +
> > +# run pylint command - For some strange reason, message controls are not taken correctly from cfg
> > +pylint --rcfile=${PYLINT_CONFIG_DIR}/travis_run_pylint.cfg --disable=all --enable=E0101,E1124,E1306,E1601,I0013,W0101,W0102,W0104,W0105,W0109,W0403,W0404,W1111,W1401 $GIT_DIR
>
> That's why I did before touch $GIT_DIR/__init__.py, because a strange
> behaviour documented in the link. Anyways, have you updated pylint and
> astroid from pip?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OCA/maintainer-quality-tools/pull/166/files#r25594148>
> .
>



-- 
*Laurent Mignon*
*Senior Software Engineer*

*Tel : +352 20 21 10 20 32*
*Fax : +352 20 21 10 21*
*Gsm : +352 691 506 009*
*Email: laurent.mignon@acsone.eu <laurent.mignon@acsone.eu>*

*Acsone SA, Succursale de Luxembourg*
*22, Zone industrielle*
*L-8287 Kehlen, Luxembourg*
*www.acsone.eu <http://www.acsone.eu>*